src/main/python/ccrf/prepare_training_data.py

```diff
@@ -10,34 +10,52 @@
 
 from utils import TfidfTgzReader
 
-topic_list = [
-    '321', '336', '341',
-    '347', '350', '362',
-    '363', '367', '375', '378', '393',
-    '397', '400', '408', '414',
-    '422', '426', '427', '433',
-    '439', '442', '445', '626', '646',
-    '690'
-]
-
-
-# all files rank file folder
-def build_docid_idx_and_label(rank_file_folder):
-    """
-    """
+# Global variables, which we're later going to refactor into an external config file.
+topics = ['321', '336', '341', '347', '350', '362', '363', '367', '375', '378',
+          '393', '397', '400', '408', '414', '422', '426', '427', '433', '439',
+          '442', '445', '626', '646', '690']
+
+config = {
+    'sources': [{'name': 'robust04',
+                 'qrels': 'src/main/resources/topics-and-qrels/qrels.robust2004.txt',
+                 'index': 'lucene-index.robust04.pos+docvectors+rawdocs'},
+                {'name': 'robust05',
+                 'qrels': 'src/main/resources/topics-and-qrels/qrels.robust2005.txt',
+                 'index': 'lucene-index.robust05.pos+docvectors+rawdocs'},
+                {'name': 'core17',
+                 'qrels': 'src/main/resources/topics-and-qrels/qrels.core17.txt',
+                 'index': 'lucene-index.core17.pos+docvectors+rawdocs'}]
+}
+
+working_directory = 'ccrf'
+
+
+def dump_docvectors():
+    for source in config['sources']:
+        qrels = source['qrels']
+        source_name = source['name']
+        index = source['index']
+        docids_file = os.path.join(working_directory, f'docids.{source_name}')
+        logging.info(f'Extracting docvectors for {source_name} with Anserini')
+        cmd = f'cut -f 3 -d " " {qrels} | sort | uniq > {docids_file}'
+        os.system(cmd)
+        cmd = f'target/appassembler/bin/IndexUtils -index {index} -dumpDocVectors {docids_file} -docVectorWeight TF_IDF'
+        os.system(cmd)
+
+
+def build_docid_idx_and_label():
     logging.info(f'start building docid idx dict and topic docid label dict')
     cur_idx = 0
     docid_idx_dict = {}
-    topic_docid_label = {topic:{} for topic in topic_list}
+    topic_docid_label = {topic: {} for topic in topics}
 
-    rank_files =['qrels.robust2004.txt', 'qrels.robust2005.txt', 'qrels.core17.txt']
-
-    for rank_file in rank_files:
+    for source in config['sources']:
+        rank_file = source['qrels']
         logging.info(f'building docid idx dict for {rank_file}')
-        with open(os.path.join(rank_file_folder, rank_file), 'r') as f:
+        with open(os.path.join(rank_file), 'r') as f:
             for line in f:
                 topic, _, docid, label = line.strip().split(' ')
-                if topic in topic_list:
+                if topic in topics:
```

Comment:
Same reason here, use `set` and get a speedup for free.

Reply:
k